engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java

```diff
@@ -232,7 +232,7 @@ public void run()
 
     String loggerAppender = dag.getValue(LogicalPlan.LOGGER_APPENDER);
     if (loggerAppender != null) {
-      vargs.add(String.format("-D%s=\"%s\"", LogicalPlan.LOGGER_APPENDER.getLongName(), loggerAppender));
+      vargs.add(String.format("-D%s='%s'", LogicalPlan.LOGGER_APPENDER.getLongName(), loggerAppender));
```

Comment:
Problem is that Yarn uses `exec /bin/bash -c "${JAVA_HOME} ..."` and $ will be treated as a special character.